### ISSUE TITLE
"Error: cannot open device with path at new HID" on Electron build

### ISSUE THREAD

**Issue by WilsonPhoo** [LedgerHQ/ledgerjs#439]
Hi,

I received this error while connecting to Nano Ledger S on a build version of my electron app.

`getUSBConnection: Error: cannot open device with path 
    at new HID (/Volumes/MyApp/MyApp.app/Contents/Resources/app.asar/node_modules/node-hid/nodehid.js:49)
    at Promise.resolve.then (TransportNodeHid.js:122)
    at <anonymous>`

I am using these packages:

dependencies:
"@ledgerhq/hw-app-eth": "^5.3.0",
"@ledgerhq/hw-transport-node-hid": "^5.3.0",
"node-hid": "^1.1.0",
"usb": "^1.6.2"

devDependencies:
"electron-builder": "^21.2.0",

building on macOS Catalina v10.15.2

Any ideas why?

Thanks!

**Comment by obiyankenobi** [comment]
I also had this problem. It's actually related to the `node-hid` lib.

I managed  to solve this following this: https://github.com/node-hid/node-hid/issues/321

**Comment by gre** [comment]
i'm not really sure if we have the same problem as discussed in https://github.com/node-hid/node-hid/issues/321
but worth an internal discussion. cc @juan-cortes @MortalKastor 

I know that this issue definitely happen if the usb connection as been acquired by another software / the device has been opened by another app like Ledger Live.

**Comment by juan-cortes** [comment]
I get that when my vm handles my nano s and I try to access it from the host machine. So the host machine can see it but when it tries to lock access to it, it fails. 🤷‍♂ Don't know if it's related.

**Comment by gre** [comment]
@juan-cortes yeah to me it's same as "some software have acquired the connection to the device". As soon as someone open() and do not close(), that error will happen on a second open() attempt.

**Comment by WilsonPhoo** [comment]
It used to work before, but stopped working after upgrading to macOS Catalina

**Comment by gre** [comment]
https://github.com/node-hid/node-hid/issues/321 seems to refer to this. I'm on Catalina and don't have any issue on Ledger Live but maybe we configured something for this.

**Comment by hoonsubin** [comment]
is there a solution to this issue yet?

**Comment by jacques-martin10** [comment]
Manual solution:

    chgrp plugdev /dev/hidraw1
    chmod 660 /dev/hidraw1

Automated solution via udev, see https://unix.stackexchange.com/a/85459
